app/wizards/schools/register_ect_wizard/wizard.rb

```diff
@@ -54,9 +55,137 @@ def self.step?(step_name)
       delegate :save!, to: :current_step
       delegate :reset, to: :ect
 
+      def allowed_steps
+        @allowed_steps ||= calculate_allowed_steps
+      end
+
+      def allowed_step?(step_name = current_step_name)
+        allowed_steps.include?(step_name)
+      end
+
       def ect
         @ect ||= ECT.new(store)
       end
+
+    private
+
+      def calculate_allowed_steps
```

Comment:
This method is quite heavy and unless I'm mistaken, it's recreating logic from the `Step` classes?

Clearly we're too invested to do anything now but there _must_ be a better way of structuring this. If you have any ideas on a better approach @edujackedu, @ltello, please raise something for the backlog.